cli/status/lua/instance_state.lua

```diff
@@ -0,0 +1,71 @@
+local ok, config = pcall(require, 'config')
+
+-- Default configuration in case the Tarantool 3.x config module is unavailable.
+local config_info_full = {
+    status = "uninitialized",
+    alerts = {},
+}
+
+-- The type check for 'config.info' ensures that we are working with the correct
+-- Tarantool 3.x config module. This is necessary because another module,
+-- 'moonlibs/config', shares the same name.
+if ok and config ~= nil and type(config) == "table" and type(config.info) == "function" then
+    config_info_full = config:info()
+end
+
+local config_info = {
+    status = config_info_full.status,
+    alerts = {}
+}
+
+-- This loop extracts only the 'type' and 'message' fields from each alert,
+-- discarding any additional fields that might be present in the config.alerts.
+-- We only need the type of the error and its message for the status report,
+-- as other fields are considered extra and unnecessary for our purposes.
+for _, alert in ipairs(config_info_full.alerts) do
+    table.insert(config_info.alerts, {
+        type = alert.type,
+        message = alert.message
+    })
+end
+
+local ok, box_replication = pcall(function() return box.info.replication end)
```

Comment:
It looks like the code hasn't been tested with Tarantool 2.
```suggestion
local ok, box_replication = pcall(function()
    if box.info.replication ~= nil then
         return box.info.replication
    end
    return box.info.cluster
end)
```

Reply:
Oh, I missed it. Thanks for catching!

I would also highlight that tarantool 2.x has no string instance name, only UUID (and tarantool 3.x may work without the name). It means that we should collect UUIDs for the `--details` report (for instances without a name).

AFAIU, it also means that matching of `tt`'s instance names to upstream names is not so straighforward and needs more complicated logic[^1]. I propose to don't attempt to match upstream/instance name on the Go side (see [here](https://github.com/tarantool/tt/pull/927#discussion_r1758674620)) and use upstream's name/UUID only for the `--details` alerts.

[^1]: Fetch `box.info.id` to the client from all the instances and create an `id2name` mapping. Then use upstream's ID and the `id2name` mapping to determine the name. I think we don't need this complication for now.

Reply:
In general, I would suggest making full support for Tarantool 3.

But for Tarantool 2 the status should be meaningful if it is not possible to support a column ​​or it is too difficult (as example, we could support alerts for Tarantool 2 + Cartridge, but that will only complicate things and maybe no one needs it, so it would be enough to create an issue).

Reply:
I agree. In future we can even make the report extensible by a role/an application. However, it seems meaningful to deliver a first version that accumulates some useful information and hear to users.

Reply:
`box.info.replication` is the same key in 2.x and 3.x. `box.info.cluster` was renamed to `box.info.replicaset`: https://github.com/tarantool/doc/issues/3465, but it is not relevant to the given code.

However, points regarding name/UUID handling are actual.

Reply:
Sorry, I was wrong. Anyway, the idea is to test and support Tarantool 2 as far as it could be done easily the same.

Reply:
> I would also highlight that tarantool 2.x has no string instance name, only UUID (and tarantool 3.x may work without the name). It means that we should collect UUIDs for the --details report (for instances without a name).

`tt` retrieves instance or application names from the configuration. The bottleneck arose from relying on `box.info.replication.[*].name`. To address this, I added a preparation step before collecting status information. This step extracts the UUID from each available instance and fills a `uuid2name` map.

For alerts printing I suggest the following:
- If `replica.UUID` exists in the `uuid2name` map => report alerts using the replica name.
- Otherwise report alerts using the replica UUID.

This ensures that we prioritize readable names when available, but fallback to UUIDs when necessary.

Reply:
Discussed in a chat with Maksim. I agree with this proposal with a few remarks:

1. It is possible that we have no tt's instance name in the mapping, but there is tarantool's instance name in `box.info.replication[<id>].upstream.name`. In this case it worth to report the tarantool's name.
2. Collecting the mapping as a separate step doubles the network requests. It seems acceptable for the initial implementation, but it would be nice to collect this information in the same eval request that grabs statuses. I would keep it as is in the scope of this PR (if other reviewers don't mind) and rework later.

Reply:
Done.